plugin-server/src/ingestion/ingestion-consumer.ts

```diff
@@ -585,7 +589,15 @@ export class IngestionConsumer {
         for (const { event, message, team } of messages) {
             const token = event.token ?? ''
             const distinctId = event.distinct_id ?? ''
-            const eventKey = `${token}:${distinctId}`
+            let eventKey: string
+            if (
+                event.properties?.[COOKIELESS_MODE_FLAG_PROPERTY] &&
+                event.properties?.[COOKIELESS_MODE_BASE_HASH_PROPERTY]
+            ) {
+                eventKey = `${token}:${event.properties[COOKIELESS_MODE_BASE_HASH_PROPERTY]}`
```

Comment:
One of the tests I added fails without this change, as the distinct ID changes part-way through the stream of events, even though they are from the same user.